src/main/java/echoserver/EchoServer.java

```diff
@@ -0,0 +1,54 @@
+package echoserver;
+
+import java.io.PrintWriter;
+import java.io.IOException;
+import java.net.Socket;
+import java.util.Scanner;
+
+public class EchoServer implements Runnable {
+    private final String QUIT = "quit";
+    private Scanner input;
+    private PrintWriter output;
+    private Socket clientSocket;
+
+    public EchoServer(Socket clientSocket) {
+        this.clientSocket = clientSocket;
+    }
+
+    public void run() {
+        try {
+            setServerInputOutput(clientSocket);
+            output.println("Connected to server");
+            echo();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void setServerIn(Scanner in) {
```

Comment:
We shouldn't be exposing public methods for the sole purpose of making this class easier to test (`#setServerIn, #setServerOut, #echo`). When we find ourselves needing to do this, it's a good indication of a code smell and that the design of our class does not allow for testability.

Ultimately, the `#run` method is the only one being used in the implementation code when we pass this runnable into a thread (line 22 in App.java) so this is the method that we should be testing through.

How can we update our design so that the only single public method in here is `#run`?